supervisor/jobs/decorator.py

```diff
@@ -61,7 +61,7 @@ async def wrapper(*args, **kwargs) -> Any:
             job = self.sys_jobs.get_job(self.name)
 
             # Handle condition
-            if self.conditions and not self._check_conditions():
+            if self.conditions and not await self._check_conditions():
```

Comment:
That's really high expensive, also since in the future we have a lot of job children. Let's split that into a function and async function and only await if there is needed